src/nvim/api/buffer.c

```diff
@@ -581,7 +581,8 @@ Object nvim_buf_get_option(Buffer buffer, String name, Error *err)
 /// @param name       Option name
 /// @param value      Option value
 /// @param[out] err   Error details, if any
-void nvim_buf_set_option(Buffer buffer, String name, Object value, Error *err)
+void nvim_buf_set_option(uint64_t channel_id, Buffer buffer,
```

Comment:
Why does the prototype need to be changed?  I don't see anything calling `nvim_buf_set_option` with the new argument.

Isn't there a way we can determine/propagate the id automatically?  If not, then there will need to be some changes to avoid breaking old API clients.

Here's an initial color for the bikeshed.  Feel free to suggest alternatives:

```c
void nvim_buf_set_option_for_channel(uint64_t channel_id, ...)
  FUNC_API_SINCE(4)
{
  ...
}

void nvim_buf_set_option(Buffer buffer, ...)
  FUNC_API_SINCE(1)
{
  nvim_buf_set_option_for_channel(0, ...);
}
```

Reply:
`uint64_t channel_id` is automatically resolved by the RPC layer, it doesn't change the signature for clients.

Reply:
Ah, ok. Thanks.

Reply:
> `uint64_t channel_id` is automatically resolved by the RPC layer, it doesn't change the signature for clients.

Is there somewhere this sort of information is documented other than in the details of the generator scripts?